09-coap: add CoAP release specs

kb2ma opened the pull request:
### Contribution description
Basic coverage for CoAP messaging as described in #73. The goal for this PR is to include non-confirmable and confirmable messaging, block extension, and observe extension. Includes some tests for both gcoap and nanocoap. Includes test of RIOT Resource Directory client implementation (CORD).

Initial commit probably premature, but I wanted to get the message out that I'm working on it for the 2018.10 release. :-)

### Issues/PRs references
Resolves #73

kb2ma commented:
Making good progress; one more task to go. I did just find a bug in nanocoap that probably should be fixed for the release, so I'll create a PR for that before continuing with the test spec.

cladmi commented:
Good to have dedicated tests using `coap` for the release.

A remark that shows a flaw in my previous naming, maybe my task 09 should be named 99 as I wanted it to be an optional bonus task. This way your task would be a more important task directly after the others. I can rename it.

jia200x commented:
thanks @kb2ma !!

kb2ma commented:
Sure thing @jia200x, glad to get this in. It gave me an excuse to work with aiocoap, which is worthwhile.

@cladmi, I'll follow your lead on the naming for the directory. If 09-compile... goes away, I'll rename 10-coap to take its place.

cladmi commented:
It went away :)

kb2ma commented:
Added Observe task, renamed to 09-coap. Squashed and ready for review. The content should be usable by others.

Also, from my perspective gcoap and nanocoap work fine in 2018.10-branch with the fix in [RIOT #10298](https://github.com/RIOT-OS/RIOT/pull/10298).

aabadie commented on 09-coap/coap.md:
```diff
@@ -0,0 +1,186 @@
+==============================
+### Description
+
+gcoap sends a CON GET requst to an aiocoap server. The test runner  delays startup of aiocoap to trigger gcoap to resend the message.
```
s/requst/request/ + 2 spaces between `runner` and `delays`

aabadie commented on 09-coap/coap.md:
```diff
@@ -0,0 +1,186 @@
+=================
+### Description
+
+aiocoap slices a payload into a sequence of Block1 encoded POSTs to nanocoap. nanocoap computes and returns the SHA-256 message digest on the payload.
```
s/on the payload/of the payload/ ?

kb2ma commented:
Pushed the wrong button. :-)

kb2ma commented:
And of course I'm happy to run these tests for the release.

kb2ma commented:
Squashed, looks good to me.

aabadie commented:
Thanks, let's merge!
